### PR TITLE
Public dashboard

### PR DESCRIPTION
**Pull request by cuducos**
This PR adds a public dashboard (no login needed) to Jarbas so people have an alternative way to browse a list of reimbursements:

It will be available at `jarbas.serenatadeamor.org/dashboard/` and no creation, update or delete actions are permitted.

### CODE REVIEW

File: jarbas/dashboard/sites.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            permission = Permission.objects.get(
+                content_type=ContentType.objects.get_for_model(Reimbursement),
+                codename='change_reimbursement'
+            )
+            user.user_permissions.add(permission)
+            user = user_model.objects.get(username='dashboard')  # avoid cache
+        return user
+
+    @staticmethod
+    def valid_url(url):
```
Comment (henriquebastos) [inline jarbas/dashboard/sites.py]: Why is this necessary? Restricting to GET isn't enougth?

File: jarbas/dashboard/sites.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    site_title = 'Dashboard'
+    site_header = 'Jarbas Dashboard'
+    index_title = 'Jarbas'
+
+    def __init__(self):
+        super().__init__('dashboard')
+        self._actions, self._global_actions = {}, {}
+
+    @staticmethod
+    def get_user():
```
Comment (henriquebastos) [inline jarbas/dashboard/sites.py]: Criar esse usuário _dashboard_ é realmente necessário? Implementar o `has_permissions` filtrando `GET` e liberando o resto não é suficiente?

File: jarbas/dashboard/templates/admin/base.html [hunk 3]
```diff
@@ -0,0 +1,71 @@
+{% load i18n static %}<!DOCTYPE html>
```
Comment (henriquebastos) [inline jarbas/dashboard/templates/admin/base.html]: Pq vc precisou replicar todos os htmls? Não dá pra saber a diferença do original olhando o commit.

**Comment by cuducos** [comment]
Many many thanks @henriquebastos!

After a hacking session with @riverfount we could get rid of the real `User` model, envolving database etc. Now an `AnonymousUser` handles the task pretty well.

TBH I also felt uncomfortable customizing three HTML files. What I would like to do is to rmeove certain elements of the UI that have no meaning in this hacked readonly loginless version:

---

pasted_image_31_05_17_19_09

---

pasted_image_31_05_17_19_14

---

Any ideas of a more effortless (thinking of medium/long term maintenance) to handle this?

**Comment by henriquebastos** [comment]
Really liked the end result.

I would tweak a couple things:

1. Simplify tests or make them more concise avoiding setup inside a test
method.
2. Replace map and filter with generator expressions.
3. Improve naming on purpose instead of content. (like the PostPut test
case that actually verifies forbidden methods).

Not that it seems necessary, just pointing out for the fun of it. :-P

**Comment by cuducos** [comment]
Many thanks again, @henriquebastos! Gonna keep that in mind and open some PR with these minor edits soon ; )
